internal/plugins/manifests/v2/api.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/afero"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+
+	"github.com/operator-framework/operator-sdk/internal/plugins/manifests/v2/internal/templates/config/samples"
+)
+
+var _ plugin.CreateAPISubcommand = &createAPISubcommand{}
+
+type createAPISubcommand struct {
+	config   config.Config
+	resource *resource.Resource
+}
+
+func (s *createAPISubcommand) InjectConfig(c config.Config) error {
+	s.config = c
+
+	// Try to retrieve the plugin config
+	if err := s.config.DecodePluginConfig(pluginKey, &Config{}); errors.As(err, &config.PluginKeyNotFoundError{}) {
+		// If we couldn't find it, it means we are not using this plugin, so we skip remaining hooks
+		return plugin.ExitError{
```

Comment:
Nit: We need to doc why we are skipping this scenario in the code. e.g add the scenario 
> “As a developer, I want to use SDK CLI for my project built with KB”
> So, in order to not fail, we will .... and the result will be ...